### ISSUE TITLE
check_overlap argument not accepted in geom_label()

### ISSUE THREAD

**Issue by larspijnappel** [tidyverse/ggplot2#4331]
The `geom_label()` is not accepting the `check_overlap` argument, although it appears to be a valid option according to the `geom_label` documentation (`ggplot2` version 3.3.3):

> Arguments `check_overlap` 	
If `TRUE`, text that overlaps previous text in the same layer will not be plotted. `check_overlap` happens at draw time and in the order of the data. Therefore data should be arranged by the label column before calling __`geom_label()`__ or __`geom_text()`__.

In another section of this documentation, there's no explicit reference that this specific argument is not supported:
> geom_label()
Currently `geom_label()` does not support the `angle` aesthetic and is considerably slower than `geom_text()`. The `fill` aesthetic controls the background colour of the label.

#### `geom_label()`
The plot is generated with a warning:
```r
mpg %>% 
  ggplot(aes(displ, hwy)) +
  geom_label(aes(label = class), check_overlap = T)
#> Warning: Ignoring unknown parameters: check_overlap
```
![image](https://user-images.githubusercontent.com/12656725/106171413-cb116880-6191-11eb-8b00-ff021558659b.png)

#### `geom_text()`
* `geom_text()` default behavior: overlapping text
```r
library(tidyverse)

mpg %>% 
  ggplot(aes(displ, hwy)) +
  geom_text(aes(label = class))
```
![image](https://user-images.githubusercontent.com/12656725/106170959-4e7e8a00-6191-11eb-9e7a-755572f46c72.png)

* `geom_text()` with `check_overlap = TRUE`: overlapping text removed
```r
mpg %>% 
  ggplot(aes(displ, hwy)) +
  geom_text(aes(label = class), check_overlap = TRUE)
```
![image](https://user-images.githubusercontent.com/12656725/106170800-22fb9f80-6191-11eb-9881-a2667057dd52.png)

**Comment by clauswilke** [comment]
Yes, the documentation is incorrect. Feel free to submit a PR to fix this.

**Comment by yutannihilation** [comment]
Fixed by #4389
